### ISSUE TITLE
Request for a MacOS GUI

### ISSUE THREAD

**Issue by godzfire** [lay295/TwitchDownloader#1267]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Write your feature request here

For those of us who are command line / programming challenged, it would be really nice if someone could please create a GUI for the Mac version to use since Windows and Linux have ones.

**Comment by superbonaci** [comment]
WPF is only for Windows, I don't know where you found a GUI for Linux. But it could work using [WINE](https://www.winehq.org/), did you check?

**Comment by superbonaci** [comment]
But that GUI is from another guy and hasn't been updated for 10 months, forget about it working correctly.
https://github.com/mohad12211/twitch-downloader-gui

I also hope someday there are GUI for all major operating systems but the CLI still needs years of work to fix all the bugs, and the CLI is the basic thing that works on all operating systems.

 The GUI is different for each operating systems and that's not including Android and iOS where even the CLI does not work.

**Comment by barclay** [comment]
Hey there gang. Thought I'd share that over the last couple of weeks I've been building [Oxbow](https://getoxbow.app/) ([repo here](https://github.com/LoftiStudios/oxbow))—it's basically an answer to this thread.

It's a native Swift/SwiftUI Mac app that drives TwitchDownloaderCLI as a subprocess. The CLI ships _inside_ the app bundle, along with .NET—pinned to a specific upstream commit (currently twelve past 1.56.5, for the 7TV emote-set endpoint migration), unmodified, with your licence and the third-party ones carried along. For the user that means: download a DMG, drag to your Applications folder, and paste a link to get a VOD. No Homebrew, no .NET, no ffmpeg (that's also built into the app), and no terminal. It does chat rendering and the side-by-side compositing as well as plain downloads—getting the signing and notarization build pipeline working was... _interesting_. (Developer ID signed and notarized, so works with the Mac's GateKeeper and opens on a double-click.)

Caveats up front: macOS 26+, Apple Silicon only. It's at 0.4.0, but treat that as early beta.

**Why a separate project rather than something upstream**: mostly because I'm deliberately building it more Mac-like, which is to say a lot more opinionated. A good number of the knobs the CLI exposes are hidden or just decided for the user... and I didn't want those choices to become annoying noise or issues for you guys. Also: I wasn't sure 20k lines of Swift and Xcode files are something you'd want in this repo anyway.

On the support side: Oxbow's bugs are mine and they go to Oxbow's tracker—I won't be sending people here. When I hit something that's genuinely a CLI issue I'll bring it upstream rather than patch around it locally ([lay295/TwitchDownloader#1644](https://github.com/lay295/TwitchDownloader/pull/1644) was the first, and I've got a couple more in the works).

Credit's where it belongs: TwitchDownloaderCLI is called out on the site, in the README, and in the app's About panel. Oxbow isn't part of TwitchDownloader proper and isn't endorsed by anyone here, but it's very much standing on your shoulders.

**Comment by godzfire** [comment]
@barclay 

> Caveats up front: macOS 26+, Apple Silicon only.

Well, sadly that eliminate me (the OP who created this thread almost 2 years ago) and a whole bunch more from the Mac community. Hopefully you've reconsider and build a UB for us Intel people and support Sequoia 15 since that's the last version a number of machines can run.

**Comment by barclay** [comment]
@godzfire 

>> Caveats up front: macOS 26+, Apple Silicon only.

> Well, sadly that eliminate me ... and a whole bunch more from the Mac community.

TBH, that was a choice to make my life easier, but it's not made out of concrete. Getting the cross-compiling working and the signing and notarization for x86 will be a ugly... but it's not impossible (I'm a little worried about ffmpeg...). But there's nothing in the project today that _requires_ Mac OS 26 (yet). 

I think the biggest issues are: 
 * I don't have any intel hardware, so I can't test it. 
 * I have no idea how good/bad the media encoders are on the old intel chips so that might make compositing the chat on a VOD a no-go on intel. 
 *  GH is sunsetting of all their Mac OS intel runners [next year](https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/#notice-of-macos-x86_64-intel-architecture-deprecation). 

Rather than dirty up this issue, [I started one](https://github.com/LoftiStudios/oxbow/issues/66) on the project. It's something I can certainly entertain.

**Comment by superbonaci** [comment]
> [@godzfire](https://github.com/godzfire)

> * I don't have any intel hardware, so I can't test it.
> * I have no idea how good/bad the media encoders are on the old intel chips so that might make compositing the chat on a VOD a no-go on intel.
> * GH is sunsetting of all their Mac OS intel runners [next year](https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/#notice-of-macos-x86_64-intel-architecture-deprecation).

I don't think you need to do any encoding, it's just stream copy.

**Comment by barclay** [comment]
@superbonaci : 

Right. And this is actually one areas where I got opinionated and diverted from the core TwitchDownoader project: I really wanted the VOD and the chat, rendered _together_. 

<img alt="Image" src="https://github.com/user-attachments/assets/91eced6a-6c97-441d-aef3-1e53d0188429" />

So that's actually one of the _default_ options for Oxbow. We're decoding the both videos, stacking them side by side into a single wider frame, and re-encode them. That's why I'm building-in ffmpeg, and have done the benchmarking of media encoders, etc... 

But you're right for just downloading a vanilla stream: absolutely not needed. 

(Also, i fully recognize this may be a frivolous and expensive feature, but it was actually the whole reason I started this silly thing. 🙃)

**Comment by superbonaci** [comment]
@barclay I also suggested to record the screen with the chat, on Twitch site while the stream is running, you do it with the web browser:

https://github.com/user-attachments/assets/d4768523-f87a-4d6c-8332-f742c667078c

The worse thing is that you have to capture while it's live from start to end, networks issues will loss content. The way you to it you can download the chat later, unless VOD storage is disabled. Also the aspect ratio is different compared to standard 16:9, but that's ok if you have an ultrawide monitor or TV.

Both ways should be captured:
- Live: captures all events, polls, and other things that are not stored in the chat json. Channels with high volume of messages may skip some.
- After stream ends: help recover from network issues or if volume of messages is too high.
